lib/couchbase-orm/base.rb

```diff
@@ -24,299 +23,302 @@
+        when CouchbaseOrm::Base
+          clear_changes_information
+          super(model.attributes.except(:id, 'type'))
+        else
+          clear_changes_information
+          assign_attributes(decode_encrypted_attributes(**attributes.merge(Hash(model)).symbolize_keys))
         end
+      else
+        clear_changes_information
+        super(attributes)
+      end
 
-        protected
+      yield self if block_given?
 
-        def serialized_attributes
-            encode_encrypted_attributes.map { |k, v|
-                [k, self.class.attribute_types[k].serialize(v)]
-            }.to_h
-        end
+      run_callbacks :initialize
     end
 
-    class NestedDocument < Document
-        def initialize(*args, **kwargs)
-            super
-            if respond_to?(:id) && id.nil?
-                assign_attributes(id: SecureRandom.hex)
-            end
-        end
+    def [](key)
+      send(key)
     end
 
-    class Base < Document
-        include ::ActiveRecord::Validations
-        include Persistence
-        include ::ActiveRecord::AttributeMethods::Dirty
-        include ::ActiveRecord::Timestamp # must be included after Persistence
-
-        include Associations
-        include Views
-        include QueryHelper
-        include N1ql
-        include Relation
-
-        extend Join
-        extend Enum
-        extend EnsureUnique
-        extend HasMany
-        extend Index
-        extend IgnoredProperties
-
-        define_model_callbacks :create, :destroy, :save, :update
-
-        class << self
-            def connect(**options)
-                @bucket = BucketProxy.new(::MTLibcouchbase::Bucket.new(**options))
-            end
-
-            def bucket=(bucket)
-                @bucket = bucket.is_a?(BucketProxy) ? bucket : BucketProxy.new(bucket)
-            end
+    def []=(key, value)
+      send(:"#{key}=", value)
+    end
 
-            def bucket
-                @bucket ||= BucketProxy.new(Connection.bucket)
-            end
+    protected
 
-            def cluster
-                Connection.cluster
-            end
+    def serialized_attributes
+      encode_encrypted_attributes.map { |k, v|
+        [k, self.class.attribute_types[k].serialize(v)]
+      }.to_h
+    end
+  end
+
+  class NestedDocument < Document
+    def initialize(*args, **kwargs)
+      super
+      if respond_to?(:id) && id.nil?
+        assign_attributes(id: SecureRandom.hex)
+      end
+    end
+  end
+
+  class Base < Document
+    include ::ActiveRecord::Validations
+    include Persistence
+    include ::ActiveRecord::AttributeMethods::Dirty
+    include ::ActiveRecord::Timestamp # must be included after Persistence
+
+    include Associations
+    include Views
+    include QueryHelper
+    include N1ql
+    include Relation
+
+    extend Join
+    extend Enum
+    extend EnsureUnique
+    extend HasMany
+    extend Index
+    extend IgnoredProperties
+    prepend JsonSchemaLoader
```

Comment:
Isn't it working with `include` instead of `prepend` ?
It's a more usual and simple way to have methods defined in a module to be available as instance methods of a class.

Reply:
I forget to replace it. I fix it by an extend